test/unit/org/apache/cassandra/tools/StandaloneVerifierOnSSTablesTest.java

```diff
@@ -0,0 +1,219 @@
+        assertEquals(0, tool.getExitCode());
+        assertCorrectEnvPostTest();
+        tool.assertOnCleanExit();
+    }
+
+    @Test
+    public void testCorruptStatsFile() throws Exception
+    {
+        String keyspaceName = "StandaloneVerifierTestCorruptStatsKs";
+        String corruptStatsTable = "corruptStatsTable";
+        createAndPopulateTable(keyspaceName, corruptStatsTable, cfs -> {
+            SSTableReader sstable = cfs.getLiveSSTables().iterator().next();
+            try (RandomAccessFile file = new RandomAccessFile(sstable.descriptor.filenameFor(Component.STATS), "rw"))
+            {
+                file.seek(0);
+                file.writeBytes(StringUtils.repeat('z', 2));
+            }
+        });
+
+        ToolResult tool = ToolRunner.invokeClass(StandaloneVerifier.class, keyspaceName, corruptStatsTable);
+
+        assertEquals(1, tool.getExitCode());
+        Assertions.assertThat(tool.getStderr()).contains("Error Loading", corruptStatsTable);
+    }
+
+    @Test
+    public void testCorruptDataFile() throws Exception
+    {
+        String keyspaceName = "StandaloneVerifierTestCorruptDataKs";
+        String corruptDataTable = "corruptDataTable";
+
+        createAndPopulateTable(keyspaceName,corruptDataTable, cfs -> {
+            SSTableReader sstable = cfs.getLiveSSTables().iterator().next();
+            long row0Start = sstable.getPosition(PartitionPosition.ForKey.get(ByteBufferUtil.bytes("0"), cfs.getPartitioner()), SSTableReader.Operator.EQ).position;
+            long row1Start = sstable.getPosition(PartitionPosition.ForKey.get(ByteBufferUtil.bytes("1"), cfs.getPartitioner()), SSTableReader.Operator.EQ).position;
+            long startPosition = Math.min(row0Start, row1Start);
+
+            try (RandomAccessFile file = new RandomAccessFile(sstable.getFilename(), "rw"))
+            {
+                file.seek(startPosition);
+                file.writeBytes(StringUtils.repeat('z', 2));
+            }
+        });
+
+        ToolResult tool = ToolRunner.invokeClass(StandaloneVerifier.class, keyspaceName, corruptDataTable);
+        assertEquals(1, tool.getExitCode());
+        Assertions.assertThat(tool.getStdout()).contains("Invalid SSTable", corruptDataTable);
+    }
+
+    /**
+     * Since we are testing a verifier, we'd like to corrupt files to verify code paths
+     * This function definition is used by {@link this#createAndPopulateTable}.
+     *
+     * CFS is the open ColumnFamilyStore for a given keyspace, table
+     */
+    private interface CorruptFunction {
+        public void apply(ColumnFamilyStore cfs) throws Exception;
+    }
+
+    /**
+     * This function sets up the keyspace, and table schema for a standardCFMD table.
+
+     * This will also populate the tableName with a few rows.  After completion the
+     * server will be shutdown.
+     *
+     * @param keyspace name the table should be created in
```

Comment:
Nit: Maybe something like:
```suggestion
     * @param keyspace the name of the keyspace in which the table should be created
```